src/ComplexType.php

```diff
@@ -117,7 +117,11 @@ protected function generateClass()
                 $getterCode = '  if ($this->' . $name . ' == null) {' . PHP_EOL
                     . '    return null;' . PHP_EOL
                     . '  } else {' . PHP_EOL
-                    . '    return \DateTime::createFromFormat(\DateTime::ATOM, $this->' . $name . ');' . PHP_EOL
+                    . '    try {' . PHP_EOL
+                    . '      return new \DateTime($this->' . $name . ');' . PHP_EOL
+                    . '    } catch (\Exception $e) {' . PHP_EOL
+                    . '      return null;' . PHP_EOL
```

Comment:
To maintain the current API this should return `false` instead of `null` like [`\DateTime::createFromFormat()` does](http://php.net/manual/en/datetime.createfromformat.php#refsect1-datetime.createfromformat-returnvalues).